### PR TITLE
Add overview for container concept

### PR DESCRIPTION
**Pull request by sftim**
Add a [page](https://deploy-preview-18090--kubernetes-io-master-staging.netlify.com/docs/concepts/containers/overview/) introducing the container concept. ([preview](https://deploy-preview-18090--kubernetes-io-master-staging.netlify.com/docs/concepts/containers/))

/kind feature

### CODE REVIEW

File: content/en/docs/concepts/containers/overview.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+and default values for any essential settings.
+
+By design, a container is immutable: you cannot change the code of a
+container that is already running. If you have a containerized application
+and want to make changes, you need to build a new container that includes
+the change, then recreate the container to start from the updated image.
+
+## Container runtimes
+
+{{< glossary_definition term_id="container-runtime" length="all" prepend="A container runtime is">}}
```
Comment (tengqm) [inline content/en/docs/concepts/containers/overview.md]: The preview for this is no good.

Comment (sftim) [reply content/en/docs/concepts/containers/overview.md]: Oops, I'd seen that and then forgot to fix it.
Thanks @tengqm!

**Comment by sftim** [comment]
Slightly related to PR #4502

**Comment by sftim** [comment]
Slightly related to PR #18335 as well.

**Comment by sftim** [comment]
Also see issue #18381

**Comment by sftim** [comment]
@thockin are you free to review this one (Eric Tune is marked busy)?

**Comment by sftim** [comment]
(I copied the reviewers from the page at https://k8s.io/docs/concepts/containers/)

**Pull request merged**